src/main/java/com/whalespottingjava/services/PostCodeService.java

```diff
@@ -0,0 +1,78 @@
+package com.whalespottingjava.services;
+
+import java.io.IOException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import org.json.JSONObject;
+
+public class PostCodeService {
+    private String urlPostCodeApi = "https://api.postcodes.io/postcodes";
```

Comment:
These fields can be final:

```suggestion
    private final String urlPostCodeApi = "https://api.postcodes.io/postcodes";
```

Reply:
These should also probably be cased as constants (UPPER_SNAKE_CASE):

```suggestion
    private String URL_POST_CODE_API = "https://api.postcodes.io/postcodes";
```